Content.Server/StationEvents/Metric/CombatMetric.cs

```diff
@@ -0,0 +1,113 @@
+///   Friend : -10 per each friendly entity on the station (negative is GOOD in chaos)
+///   Hostile : about 10 points per hostile (those with antag roles) - varies per constants
+///   Combat: friendlies + hostiles (to represent the balance of power)
+///   Death: 20 per dead body,
+///   Medical: 10 for crit + 0.05 * damage (so 5 for 100 damage),
+/// </summary>
+public sealed class CombatMetric : ChaosMetricSystem<CombatMetricComponent>
+{
+    public override ChaosMetrics CalculateChaos(EntityUid metric_uid, CombatMetricComponent component, ChaosMetricComponent metric,
+        CalculateChaosEvent args)
+    {
+        // Add up the pain of all the puddles
+        var query = EntityQueryEnumerator<MindComponent, MobStateComponent, DamageableComponent>();
+        FixedPoint2 hostiles = 0.0f;
+        FixedPoint2 friendlies = 0.0f;
+
+        FixedPoint2 medical = 0.0f;
+        FixedPoint2 death = 0.0f;
+
+        var nukieQ = GetEntityQuery<NukeOperativeComponent>();
+        var zombieQ = GetEntityQuery<ZombieComponent>();
+
+        var humanoidQ = GetEntityQuery<HumanoidAppearanceComponent>();
+
+        while (query.MoveNext(out var uid, out var mind, out var mobState, out var damage))
+        {
+            // Don't count anything that is mindless
+            if (mind.Mind == null)
+                continue;
+
+            if (mind.Mind.HasAntag)
+            {
+                if (mobState.CurrentState != MobState.Alive)
+                    continue;
+
+                // This is an antag
+                if (nukieQ.TryGetComponent(uid, out var nukie))
+                {
+                    hostiles += component.HostileScore + component.NukieScore;
+                }
+                else if (zombieQ.TryGetComponent(uid, out var zombie))
+                {
+                    hostiles += component.HostileScore + component.ZombieScore;
+                }
+                else
+                {
+                    hostiles += component.HostileScore;
+                }
+            }
+            else
+            {
+                // This is a friendly
+                // Quick filter for non-pets
+                if (!humanoidQ.TryGetComponent(uid, out var humanoid))
+                {
+                    continue;
+                }
+
+                if (mobState.CurrentState == MobState.Dead)
+                {
+                    death += component.DeadScore;
+                    continue;
+                }
+                else
+                {
+                    medical += damage.Damage.Total * component.MedicalMultiplier;
+                    if (mobState.CurrentState == MobState.Critical)
+                    {
+                        medical += component.CritScore;
+                        continue;
+                    }
+                }
+
+                // Friendlies are good, so make a negative chaos score
+                friendlies -= component.FriendlyScore;
+
+            }
```

Comment:
As above this is like 4 different systems combined into 1.

Reply:
Yes, I'll split these. I was just reluctant to make multiple metrics which were each iterating all players.